airbyte-integrations/connectors/source-instagram/source_instagram/client/api.py

```diff
@@ -330,28 +330,21 @@ def list(self, fields: Sequence[str] = None) -> Iterator[dict]:
             ig_account = account["instagram_business_account"]
             media = self._get_media(ig_account, {"limit": self.result_return_limit}, ["media_type"])
             for ig_media in media:
-                try:
-                    yield {
-                        **{
-                            "id": ig_media.get("id"),
-                            "page_id": account["page_id"],
-                            "business_account_id": ig_account.get("id"),
-                        },
-                        **{record.get("name"): record.get("values")[0]["value"] for record in self._get_insights(ig_media)},
-                    }
-                except FacebookRequestError as error:
-                    # An error might occur if the media was posted before the most recent time that
-                    # the user's account was converted to a business account from a personal account
-                    if error.api_error_subcode() == 2108006:
-                        logger.error(f"Insights error for business_account_id {ig_account.get('id')}: {error.body()}")
-
-                        # We receive all Media starting from the last one, and if on the next Media we get an Insight error,
-                        # then no reason to make inquiries for each Media further, since they were published even earlier.
-                        break
-                    raise error
+                account_id = ig_account.get("id")
+                media_insights = self._get_insights(ig_media, account_id)
+                if media_insights is None:
+                    break
+                yield {
+                    **{
+                        "id": ig_media.get("id"),
+                        "page_id": account["page_id"],
+                        "business_account_id": account_id,
+                    },
+                    **{record.get("name"): record.get("values")[0]["value"] for record in media_insights},
```

Comment:
Why we use `get(key)` method instead of d[key] ?
If record.get("values")` = None, then record.get("values")[0] will cause en exception.
Same for the records above, if they should not be `None`.